### ISSUE TITLE
Some Input Events are fired more than once or without a cause

### ISSUE THREAD

**Issue by quadratron** [opentk/opentk#488]
### Description

MouseWheel Events fire twice even when the scroll wheel is only turned one step.
Also left- or rightclicking in the window causes not only the MouseUp / MouseDown Events
to fire but also fires MouseEnter and MouseLeave Events.

#### MouseWheel Event

(Only works with stepped scroll wheel)

1. Create a new GameWindow
2. Subscribe to MouseWheel Event and Log calls
3. Move scroll wheel in GameWindow

#### MouseUp / MouseDown Event

1. Create a new GameWindow
2. Subscribe to MouseUp, MouseDown, MouseEnter, MouseLeave Events and Log calls
3. Leftclick in GameWindow

#### Sample Code

```c-sharp
using System;
using OpenTK;

namespace OpenTKInputTest
{
	class Program
	{
		static void Main(string[] args)
		{
			GameWindow window = new GameWindow();
			window.MouseEnter += (o, e) => { Console.WriteLine("[{0}] Mouse entered window", GetHwnd(o)); };
			window.MouseLeave += (o, e) => { Console.WriteLine("[{0}] Mouse left window", GetHwnd(o)); };
			window.MouseDown += (o, e) => { Console.WriteLine("[{0}] {1} {2}", GetHwnd(o), e.Button, e.IsPressed); };
			window.MouseUp += (o, e) => { Console.WriteLine("[{0}] {1} {2}", GetHwnd(o), e.Button, e.IsPressed); };
			window.MouseWheel += (o, e) => { Console.WriteLine("[{0}] {1}", GetHwnd(o), e.DeltaPrecise); };

			window.Run(60.0);
		}

		static string GetHwnd(object window)
		{
			NativeWindow win = window as NativeWindow;
			return win.WindowInfo.Handle.ToString("X8");
		}
	}
}
```
#### Call Stack in second MouseWheel Event

```
>	OpenTKInputTest.exe!OpenTKInputTest.Program.Main.AnonymousMethod__0_4(object o, OpenTK.Input.MouseWheelEventArgs e) Line 17	C#
 	OpenTK.dll!OpenTK.NativeWindow.OnMouseWheel(OpenTK.Input.MouseWheelEventArgs e)	Unknown
 	OpenTK.dll!OpenTK.NativeWindow.OnMouseWheelInternal(object sender, OpenTK.Input.MouseWheelEventArgs e)	Unknown
 	OpenTK.dll!OpenTK.Platform.NativeWindowBase.OnMouseWheel(float dx, float dy)	Unknown
 	OpenTK.dll!OpenTK.Platform.Windows.WinGLNative.HandleMouseWheel(System.IntPtr handle, OpenTK.Platform.Windows.WindowMessage message, System.IntPtr wParam, System.IntPtr lParam)	Unknown
 	OpenTK.dll!OpenTK.Platform.Windows.WinGLNative.WindowProcedure(System.IntPtr handle, OpenTK.Platform.Windows.WindowMessage message, System.IntPtr wParam, System.IntPtr lParam)	Unknown
 	[Native to Managed Transition]	
 	[Managed to Native Transition]	
 	OpenTK.dll!OpenTK.Platform.Windows.WinGLNative.WindowProcedure(System.IntPtr handle, OpenTK.Platform.Windows.WindowMessage message, System.IntPtr wParam, System.IntPtr lParam)	Unknown
 	[Native to Managed Transition]	
 	[Managed to Native Transition]	
 	OpenTK.dll!OpenTK.Platform.Windows.WinGLNative.ProcessEvents()	Unknown
 	OpenTK.dll!OpenTK.NativeWindow.ProcessEvents(bool retainEvents)	Unknown
 	OpenTK.dll!OpenTK.GameWindow.Run(double updates_per_second, double frames_per_second)	Unknown
 	OpenTK.dll!OpenTK.GameWindow.Run(double updateRate)	Unknown
 	OpenTKInputTest.exe!OpenTKInputTest.Program.Main(string[] args) Line 20	C#
```

### Expected behavior

MouseWheel Event fires exactly once for every scroll wheel step.
MouseUp / MouseDown Event does NOT fire MouseEnter / MouseLeave Events.

### Actual behavior

MouseWheel Event fires two times for every scroll wheel step.
MouseUp / MouseDown Event DOES fire MouseEnter / MouseLeave Events.

### Related information 

* Windows 10 Pro 1607
* Native Backend
* OpenTK latest Nuget, latest Repo
* .Net Framework 4.5.2

**Comment by croxxx** [comment]
I have run into the issue with the **MouseWheel** event handler as well.

### Affected Platforms
Currently this seems to be a Windows 10 exclusive problem.
When using Linux or Windows 7 with the same code the MouseWheel event is not fired twice.

_It is worth noting that this only affects the MouseWheel event. The other events named in the previous comment do not change their behaviour._

### Workaround
A workaround is to disable the "allow scrolling inactive windows" option as described here: https://www.tenforums.com/tutorials/12816-turn-off-scroll-inactive-windows-windows-10-a.html

Of course this is no permanent solution but it might be a hint that the issue is caused by an internal change that occured in Windows 10.

### System information
- Windows 10 Pro 1607 (Build 14393.969)
- Latest OpenTK from NuGet (2.0.0). The issue was present in earlier versions, too.
- .NET Framework 4.5.2

### Code used for testing
```csharp
using System;
using OpenTK;

namespace opentk_eventtest
{
    class Program
    {
        static void Main(string[] args)
        {
            Program p = new Program();
        }

        public Program()
        {
            GameWindow window = new GameWindow(1280, 720);
            window.MouseWheel += Window_MouseWheel;
            window.VSync = VSyncMode.On;
            window.Run();
        }

        private void Window_MouseWheel(object sender, OpenTK.Input.MouseWheelEventArgs e)
        {
            Console.WriteLine("Event received");
            Console.WriteLine("Delta {0}", e.Delta);
            Console.WriteLine("DeltaPrecise {0}", e.DeltaPrecise);
            Console.WriteLine("Value {0}", e.Value);
            Console.WriteLine("ValuePrecise {0}", e.ValuePrecise);
        }
    }
}

```

**Comment by UnknownShadow200** [comment]
@varon 
Not fully fixed - I still get Leave/Enter events for every click on windows 10.

**Comment by varon** [comment]
So you are correct. That's incredibly odd.

**Comment by jeske** [comment]
my patch fixes the spurious mouse enter/leave messages

**Comment by varon** [comment]
fixed by #539
